Add code style check and build script

patrickrobrecht opened the pull request:
- Adds a code style check for CSS files using stylelint. Existing violations are resolved already.
- Adds a code style check for PHP files. Existing violations are fixed.
- Defines a build script for creating the `.min.css` files. The currently versioned `.min.css` files are removed.
- Changes the GitHub Actions to include a `composer install`. In addition, the branch is changed from `master` to `stable`. Before the next release we should rename `master` to `develop` and create a new `stable` branch.
- Add .distignore to deply minified CSS files and to ignore README.md (in facor of readme.txt)

patrickrobrecht committed:
Configure code style checks for CSS (with stylelint, existing violations fixed) and PHP (with CodeSniffer, existing violations only fixed if that works automatically, exceptions for all other rules); setup minification of CSS files and remove versioned min.css files

patrickrobrecht committed:
Change GitHub Actions: add build via composer, prepare new branch strategy

patrickrobrecht committed:
Resolve PHP code style violations

patrickrobrecht committed:
Resolve PHP code style violations: class names and css files enqueued with plugin version

patrickrobrecht committed:
Resolve PHP code style violations: escape output and sanitize input

patrickrobrecht committed:
Merge branch 'master' into build

patrickrobrecht committed:
Add changelog

patrickrobrecht committed:
Update dependencies, including switch from stylelint-config-wordpress to @wordpress/stylelint-config

stklcode commented on .github/workflows/wordpress-plugin-asset-update.yml:
```diff
@@ -2,16 +2,17 @@ name: Plugin asset/readme update
 on:
   push:
     branches:
```
We obviously habe to rename the branches before this works.

patrickrobrecht replied:
My plan is to create the `develop` branch directly after this PR is merged and `stable` when the next release is ready.

stklcode commented on .github/workflows/wordpress-plugin-asset-update.yml:
```diff
@@ -2,16 +2,17 @@ name: Plugin asset/readme update
     steps:
-    - uses: actions/checkout@master
+    - uses: actions/checkout@v2
+    - uses: php-actions/composer@v1
```
Any reason for v1?

stklcode commented:
We do use “GPLv3 or later“ consistently, also in the new _composer.json_, but the License URI points to GPL 2.0.

Domain path should also be redundant as we removed the language files.

And the GH branch no longer matches the deployment branch from the action.

https://github.com/pluginkollektiv/snitch/blob/07db7176aebbc7dbdc3e1c86e5b234cae8fb1b78/snitch.php#L8-L24

patrickrobrecht committed:
Use latest composer action version in GitHub actions, make license URI consistent between readme.txt and main plugin file, correct some spelling mistakes

patrickrobrecht commented:
> We do use “GPLv3 or later“ consistently, also in the new _composer.json_, but the License URI points to GPL 2.0.

Fixed.

> Domain path should also be redundant as we removed the language files.

Removed, should have been in the PR #40.
 
> And the GH branch no longer matches the deployment branch from the action.
> 
> https://github.com/pluginkollektiv/snitch/blob/07db7176aebbc7dbdc3e1c86e5b234cae8fb1b78/snitch.php#L8-L24

The two GitHub lines are removed completely.

stklcode reviewed (changes requested):
Good work.

Think we do need a _.distignore_ for a fully functional workflow, though to get the minified resourced deployed and to not deploy the _README.md_

patrickrobrecht committed:
Add .distignore to deply minified CSS files and to ignore README.md (in facor of readme.txt)

patrickrobrecht commented:
> Think we do need a _.distignore_ for a fully functional workflow, though to get the minified resourced deployed and to not deploy the _README.md_

You are right, addressed with the latest commit.

stklcode reviewed (approved):
LGTM
